sotopia/envs/evaluators.py

```diff
@@ -150,6 +150,98 @@ def minus_ten_to_zero_validator(cls, v: tuple[str, int]) -> tuple[str, int]:
         return v
 
 
+class EvaluationByGroup(BaseModel):
+    effectiveness: tuple[str, int] = Field(
+        ...,
+        description="""Consider features of verbosity (e.g. word count, sentence lengths, readability)
+         to analyze the interaction episode between the agents and use this as a method in evaluating 
+         the efficiency of this episode. Higher levels of efficiency should be reflected with focused, 
+         coherent, and collaborative communication that advances the team's objectives. Lower efficiency 
+         can be marked by disjointed, repetitive, or ineffective exchanges that hinder or delay the progress
+         in attaining the goal."""
+        """Output your reasoning process to the 'reasoning' field. Output an integer score ranging from 0 and 
+        10 in the 'score' field. A higher score indicates that the agent is more efficient.""",
+    )
+    equity: tuple[str, int] = Field(
+        ...,
+        description="""Analyze the script of a team interaction and measure verbal equity, 
+        i.e., the extent to which the interaction shows a balanced distribution of speaking 
+        opportunities among team members. High verbal equity occurs when all team members 
+        have an equal chance to contribute their ideas, perspectives, and knowledge, and low 
+        verbal equity happens when the conversation tends to be dominated by one or a few individuals."""
+        """Output your reasoning process to the 'reasoning' field. Output an integer score ranging from 0 and 
+        10 in the 'score' field. A higher score indicates a higher level of verbal equity.""",
+    )
+    transactivity: tuple[str, int] = Field(
+        ...,
+        description="""Analyze the provided social interaction episode between the given pair/team, 
+        focusing on identifying instances of transactive exchanges. High transactivity occurs when 
+        interlocutors actively elaborate on, build upon, or argue against previously presented ideas, 
+        fostering dynamic exchanges and deeper engagement. In contrast, low transactivity occurs when 
+        interactions involve minimal elaboration or engagement with prior contributions, resulting in 
+        less dynamic and interactive exchanges."""
+        """Output your reasoning process to the 'reasoning' field. Output an integer score ranging from 0 and 
+        10 in the 'score' field. A higher score indicates a higher level of transactivity.""",
+    )
+
+    @validator("effectiveness", "equity", "transactivity")
+    def zero_to_ten_validator(cls, v: tuple[str, int]) -> tuple[str, int]:
+        assert v[1] >= 0 and v[1] <= 10
+        return v
+    
+
+class EvaluationByGroupPlus(BaseModel):
```

Comment:
# TODO: 
check if this is working better than the `EvaluationByGroup`, if not, consider removing it.